lang/py/avro/schema.py

```diff
@@ -221,6 +232,18 @@ def to_json(self, names):
         """
         raise Exception("Must be implemented by subclasses.")
 
+    def validate(self, datum):
```

Comment:
Can this be an abstractmethod? If you don't want to do it in this PR I can make all these de-facto abstract methods official in a later batch change.

Reply:
in other comments we've decided to leave this work for another PR.  Resolving.